students/laud/session04/kata_fourteen.py

```diff
@@ -0,0 +1,63 @@
+#!/usr/bin/env python3
+import sys
+import random
+import string
+
+# Strip out punctuations, etc
+def cleanup(text):
+    lowercase = text.lower()
+    exclude = set(string.punctuation)
+    out = ''.join( ch for ch in lowercase if ch not in exclude)
+    return out
+
+# Read txt file
+def read_in_data(filename):
+    with open(filename, 'r') as text:
+        data = text.read()
+        clean = cleanup(data)
+    return clean
+
+# Split text into list
+def make_words(in_data):
+    data = in_data.split()
+    return data
+
+# Build key-value pairs
+def build_trigram(words):
+    trigrams = {}
+    for i in range(len(words)-2):
+        pair = tuple(words[i:i + 2])
+        follower = words[i + 2]
+        if pair not in trigrams:
+           trigrams[pair] = []
+        trigrams[pair].append(follower)
```

Comment:
Minor: Lines 31-33 can be simplified to `trigrams.setdefault(pair, []).append(follower)`, or even define `trigrams = collections.defaultdict(list)` (after `import collections` at the top of this source file.

Reply:
@hosungs Thanks for the feedback. I'll attempt to fix these PRs, but yes, we can take a look at them in class. Remind me, though, isn't it this Tuesday that you're going to be out of town?

Reply:
@hosungs I just remembered how this happened! I think I accidentally ran `git pull upstream master` before committing my stuff. I'll just go ahead and remove the stuff in the other students' folders, and then re-commit. I think that should do it.

Reply:
@hosungs Also, should I just go ahead and maintain this one PR for all my late assignments, instead of attempting to create a separate PR for Session03 and Session04? Seems to me that would be less messy.

Reply:
> @hosungs Also, should I just go ahead and maintain this one PR for all my late assignments, instead of attempting to create a separate PR for Session03 and Session04? Seems to me that would be less messy.

Yes, we should be able to do with one single PR. Thanks.